ably/types/presence.py

```diff
@@ -85,6 +85,69 @@ def member_key(self):
     def extras(self):
         return self.__extras
 
+    def is_synthesized(self):
+        """
+        Check if message is synthesized (RTP2b1).
+        A message is synthesized if its connectionId is not an initial substring of its id.
+        This happens with synthesized leave events sent by realtime to indicate
+        a connection disconnected unexpectedly.
+        """
+        if not self.id or not self.connection_id:
+            return False
+        return not self.id.startswith(self.connection_id + ':')
+
+    def parse_id(self):
+        """
+        Parse id into components (connId, msgSerial, index) for RTP2b2 comparison.
+        Expected format: connId:msgSerial:index (e.g., "aaaaaa:0:0")
+
+        Returns:
+            dict with 'msgSerial' and 'index' as integers
+
+        Raises:
+            ValueError: If id is missing or has invalid format
+        """
+        if not self.id:
+            raise ValueError("Cannot parse id: id is None or empty")
+
+        parts = self.id.split(':')
+        if len(parts) < 3:
+            raise ValueError(f"Cannot parse id: expected format 'connId:msgSerial:index', got '{self.id}'")
+
+        try:
+            return {
+                'msgSerial': int(parts[1]),
+                'index': int(parts[2])
+            }
+        except (ValueError, IndexError) as e:
+            raise ValueError(f"Cannot parse id: invalid msgSerial or index in '{self.id}'") from e
+
+    def to_encoded(self, cipher=None):
```

Comment:
could you take a look at the method `as_dict` in the regular message? there are some additional steps when we serialize message to json, if data is binary we need to add `'/base64'` to encoding and encode data to string.

Also we don't encrypt data right now

Reply:
i've fixed the encoding and also added support for encryption